src/utils/index.test.ts

```diff
@@ -25,54 +26,209 @@ describe('Utils', () => {
 
     it('should construct a HTML element with default attributes and values', () => {
       const element = 'lite-element';
-      const attrs = {
+      const defaultAttrs = {
         id: '123',
         loading: 'lazy',
       };
 
-      const htmlElement = createHtml(element, attrs);
+      const htmlElement = createHtml(element, defaultAttrs);
       expect(htmlElement).toEqual(
         '<lite-element id="123" loading="lazy"></lite-element>',
       );
     });
 
     it('should construct a HTML element passing parameters to any required src URLs', () => {
       const element = 'lite-element';
-      const attrs = {
+      const defaultAttrs = {
         id: '123',
         src: {
           url: 'https://example.com/',
           params: ['unit', 'type'],
         },
       };
-      const args = {
+      const urlQueryParamInputs = {
         unit: 'imperial',
         type: 'main',
       };
 
-      const htmlElement = createHtml(element, attrs, args);
+      const htmlElement = createHtml(
+        element,
+        defaultAttrs,
+        {},
+        urlQueryParamInputs,
+      );
       expect(htmlElement).toEqual(
         '<lite-element id="123" src="https://example.com/?unit=imperial&type=main"></lite-element>',
       );
     });
 
     it('should construct a HTML element overwriting default attribute values with user-defined inputs', () => {
       const element = 'lite-element';
-      const attrs = {
+      const defaultAttrs = {
         id: '123',
         src: {
           url: 'https://example.com/',
           params: ['unit', 'type'],
         },
       };
-      const args = {
+      const htmlAttrInputs = {
         src: 'https://example.com/overwrite',
       };
 
-      const htmlElement = createHtml(element, attrs, args);
+      const htmlElement = createHtml(element, defaultAttrs, htmlAttrInputs);
       expect(htmlElement).toEqual(
         '<lite-element id="123" src="https://example.com/overwrite"></lite-element>',
       );
     });
   });
+
+  describe('formatData', () => {
+    it('should correctly format and overwrite data and inputs', () => {
+      const data = {
+        id: 'third-party',
+        description: 'Description',
+        html: {
+          element: 'iframe',
+          attributes: {
+            loading: 'lazy',
+            src: {
+              url: 'https://www.example.com/',
+              params: ['id'],
+            },
+            width: '100',
+            height: '100',
+          },
+        },
+      };
+
+      const inputs = {
+        id: 'props.id',
+        loading: 'auto',
+        width: '150',
+      };
+
+      const result = formatData(data, inputs);
+      expect(result.html).toEqual(
+        '<iframe loading="auto" src="https://www.example.com/?id=props.id" width="150" height="100"></iframe>',
```

Comment:
I'm not sure if this is right `src="https://www.example.com/?id=props.id"` since it appends `id=props.id` and I was thinking it would be like `id={props.id}`
I think I might be misunderstanding these tests and we don't have to do this. I can test the actual api once updated and revisit this later.

Reply:
Yeah let me merge this in for you so you can test it out